lib/aca_entities/protocols/transmittable/process_status.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module AcaEntities
+  module Protocols
+    module Transmittable
+      # The current state of a process and the history of its state transitions
+      class ProcessStatus < Dry::Struct
```

Comment:
this is a question; this looks slightly similar to workflow_state_transition.rb, but the other one is more targeted for state machine, is the purpose here to don't create a state machine ?

Also I may be misunderstanding the whole thing, but the comment I'm going is why not re-use the transitions object, or maybe the purpose is to reflect a state in time and not hold all the states

Reply:
We do want all the states for history tracking, while process status is the current view with the history available. This was a pattern designed by Dan that we implemented.